core/src/main/java/org/mule/TransformationService.java

```diff
@@ -429,19 +427,29 @@ private void transformMessage(MuleMessage message, MuleEvent event, Transformer
                 // instance
                 synchronized (message)
                 {
-                    MuleMessage resultMessage = (MuleMessage) result;
-                    message.setPayload(resultMessage.getPayload(), resultMessage.getDataType());
-                    message.setOriginalPayload(resultMessage.getOriginalPayload());
-                    message.copyMessageProperties(resultMessage);
-                    message.copyAttachments(resultMessage);
+                    MuleMessage transformResult = (MuleMessage) result;
```

Comment:
I think this can be simplified now as the DefaultMuleMessage constructor does this.